packages/typegpu-wgsl-parser/tests/structDecl.spec.ts

```diff
@@ -0,0 +1,204 @@
+                  type: 'attribute',
+                  ident: 'size',
+                  args: [{ type: 'int_literal', value: '32' }],
+                },
+              ],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'vec2',
+                template_list: [
+                  {
+                    type: 'template_elaborated_ident',
+                    ident: 'f32',
+                    template_list: null,
+                  },
+                ],
+              },
+            } satisfies StructMember,
+            {
+              type: 'struct_member',
+              ident: 'to',
+              attrs: [
+                {
+                  type: 'attribute',
+                  ident: 'align',
+                  args: [{ type: 'int_literal', value: '32' }],
+                },
+              ],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'vec3f',
+                template_list: null,
+              },
+            } satisfies StructMember,
+          ],
+        } satisfies StructDecl,
+      ],
+      type: 'translation_unit',
+    });
+  });
+
+  it('parses nested struct', () => {
+    expect(
+      parse(
+        'struct Gradient { from: vec3f, to: vec3f, }; struct Material { color: vec3f, gradient: Gradient, };',
+      ),
+    ).toEqual({
+      declarations: [
+        {
+          type: 'struct_decl',
+          ident: 'Gradient',
+          members: [
+            {
+              type: 'struct_member',
+              ident: 'from',
+              attrs: [],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'vec3f',
+                template_list: null,
+              },
+            } satisfies StructMember,
+            {
+              type: 'struct_member',
+              ident: 'to',
+              attrs: [],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'vec3f',
+                template_list: null,
+              },
+            } satisfies StructMember,
+          ],
+        } satisfies StructDecl,
+        null,
+        {
+          type: 'struct_decl',
+          ident: 'Material',
+          members: [
+            {
+              type: 'struct_member',
+              ident: 'color',
+              attrs: [],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'vec3f',
+                template_list: null,
+              },
+            } satisfies StructMember,
+            {
+              type: 'struct_member',
+              ident: 'gradient',
+              attrs: [],
+              typespec: {
+                type: 'template_elaborated_ident',
+                ident: 'Gradient',
+                template_list: null,
+              },
+            } satisfies StructMember,
+          ],
+        } satisfies StructDecl,
+        null,
```

Comment:
what are these nulls?

Reply:
They are a result of our preprocesor changing optional `;` to null:
```ne
global_decl ->
    ";" {% () => null %}  <----- Here
  | global_variable_decl ";" {% id %}
  | value_decl ";" {% id %}
  | override_decl ";" {% id %}
  | struct_decl {% id %}
  | function_decl {% id %}
  # | type_alias_decl ";"
  # | const_assert_statement ";"
```